### PR TITLE
Add World Walkers

### PR DESCRIPTION
**Pull request by Mjb141**

### CODE REVIEW

File: src/script/campaign/mod/vco-listeners.lua [hunk 1]
```diff
@@ -139,6 +193,18 @@ local function check_vco_kho_exiles_of_khorne_skarbrand_rank_40(character)
 	end
 end
 
+local function check_vco_nor_wulfrik_chaos_allegiance_levels(pooled_resource)
+  local FINISHED_ONE_GOD_ALREADY = cm:get_saved_value("vco_nor_wul_aligned_to_god") or false;
```
Comment (Mjb141) [inline src/script/campaign/mod/vco-listeners.lua:197]: Is any of this needed? Wouldn't this work if I just switched the `true` to `false` in the listener so it fires once and once only on the first occurrence of a chaos god reaching 100 pooled resource?

**Comment by Mjb141** [comment]
## What's left to implement @msolefonte :

### Route 1

#### Objectives
* Nothing

#### Rewards
* Link the effect and the listener - the listener probably needs to check if the route has been completed in it's conditional? [This ](https://github.com/msolefonte/vco/pull/199/files#diff-ae01b5f1fbe52868cbfe7c6ba2e3497349721a5fe1f82a533e8284adbd6852fdR489)will currently work from the start of the game. Alternatively find a way (if possible) to modify existing movement range effects in the db to only trigger on port settlements, which would remove the need for a listener. I doubt this is possible?

### Route 2

#### Objectives
* Nothing

#### Rewards
* Implement Grimgor's "You're Next" ability for Wulfrik.

### Route 3

#### Objectives
* Nothing

#### Rewards
* Nothing

**Comment by Mjb141** [comment]
## What's left to test @OhManTFE :

### Route 1

#### Objectives
* "Attain level 3 allegiance with a Chaos God" completes on maxing a god's pooled resource
* "Defeat all of the Challengers sent against you" completes after the challenger battles are won

#### Rewards
* Movement range subject to above implementation
* Units unlock for the first (and only the first) god that reaches level 3 for settlement building levels 3 and 4

### Route 2

#### Objectives
* Nothing

#### Rewards
* Wulfrik gets an ability and can use it

### Route 3

#### Objectives
* "Complete 6 Monster Hunts" completes after the six monster hunt final battles are won

#### Rewards
* "Primal Instincts for Marauder Spearmen and Marauder Hunters" applies to spearmen and hunters and nothing else
* "Bonus vs Large +6 for Marauder Spearmen and Marauder Hunters" applies to spearmen and hunters and nothing else
* "Enables poison attacks" applies for marauders, beserkers, champions, and nothing else

**Comment by OhManTFE** [comment]
I need a packfile to test

File: src/script/campaign/mod/vco-listeners-nor.lua [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+local UNLOCKABLE_SLAANESH_UNITS = { "wh3_main_sla_inf_daemonette_0", "wh3_main_sla_inf_daemonette_1" }
+
+local function wul_lock_units()
+	vlc.unit_locks:lock_units(UNLOCKABLE_KHORNE_UNITS, FACTION_WUL_KEY);
+	vlc.unit_locks:lock_units(UNLOCKABLE_NURGLE_UNITS, FACTION_WUL_KEY);
+	vlc.unit_locks:lock_units(UNLOCKABLE_TZEENTCH_UNITS, FACTION_WUL_KEY);
+	vlc.unit_locks:lock_units(UNLOCKABLE_SLAANESH_UNITS, FACTION_WUL_KEY);
+end
+
+local function wul_unlock_units(pooled_resource)
```
Comment (Mjb141) [inline src/script/campaign/mod/vco-listeners-nor.lua:32]: Should this be `lock_units`?

Comment (msolefonte) [reply src/script/campaign/mod/vco-listeners-nor.lua:32]: Fixed

**Comment by OhManTFE** [comment]
Praise the Omnissiah

**Review by msolefonte — APPROVED** [review]
After All, Why Not? Why Shouldn't I?

**Pull request merged**